client/state/happychat/middleware.js

```diff
@@ -67,6 +132,14 @@ export default function( connection = null ) {
 
 	return store => next => action => {
 		switch ( action.type ) {
+			case HAPPYCHAT_CONNECTING:
+				if ( getHappychatConnectionStatus( store.getState() ) === 'connected' ) {
+					// If chat is already connected, do nothing and stop the action from proceeding to reducers
+					return;
```

Comment:
I'm not to sure the best solution here is to "block" the action from hitting the reducer especially since you're not guaranteed to be the first middleware that see's and reacts to it. with that said I agree with @southp here that we should probably add the connection to a query component and have something like `<HappychatConnection />` be rendered instead of calling `connectChat` directly in a componentDidMount method of a display component. The `HappychatConnection` component can then decide whether or not it wants to dispatch the `HAPPYCHAT_CONNECT` action. Just having the `HappychatConnection` component and adequate documentation should encourage its use over dispatching the action directly.